lib/core/api.js

```diff
@@ -171,7 +171,8 @@ api.upload.accounts = function(happyCb, sadCb) {
 };
 
 function postBlockToPlatform(data, groupId, callback) {
-  if (!callback) {
+
+  if (typeof groupId === 'function') {
     callback = groupId;
```

Comment:
I thought I said this to Jamie recently, so I'm sorry if I'm repeating myself. This form of optional arguments is challenging to maintain -- it's not obvious that it's an optional argument. It's useful in an API when you have third parties that may be using it -- but in this case, all the clients are within this app. I'd rather fix up all the references and remove the optionality.